modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -518,7 +518,16 @@ public static MetadataType getFieldMetadataType(Field field, ClassTypeLoader typ
 
   public static Optional<Field> getField(Class<?> clazz, String name) {
     Collection<Field> candidates = getAllFields(clazz, withName(name));
-    return isEmpty(candidates) ? empty() : Optional.of(candidates.iterator().next());
+    if (isEmpty(candidates)) {
+      return empty();
+    }
+
+    Field field = candidates.stream()
+        .filter(f -> f.getDeclaringClass().equals(clazz))
```

Comment:
this will probably have side effects and doesn't really comply with the intent of this method. Why do you need this? It should be a separate method then (not to mention than doing a getAllFields to then filter by declaring class is very unneficient